docs/dev.rst

```diff
@@ -163,6 +163,46 @@ public by default with:
 
     $ gsutil defacl set public-read gs://my-gcs-bucket
 
+Emulated Google Cloud Storage
+=============================
+
+The ``gcs-emulator`` container provides an emulated Google Cloud Storage service
+as well as a public host for downloads. It supports the operations needed by
+Tecken, such as creating and listing buckets, uploading and reading files, and
+serving files via the public host. Further GCS features may not be supported,
+but can be added to `fsouza/fake-gcs-server`_.
+
+To use the emulator, set the URLs in your ``.env`` file:
+
+.. code-block:: shell
+
+    DJANGO_SYMBOL_URLS=https://gcs-emulator.127.0.0.1.nip.io:4443/tecken
+    DJANGO_UPLOAD_DEFAULT_URL=https://gcs-emulator.127.0.0.1.nip.io:4443/tecken
+
+The bucket (``tecken`` in this example) will be created when the web server
+starts.
+
+Tecken changes the Google Cloud Storage client libraries to use the
+``gcs-emulator`` URLs rather than the official URLs. It is not possible to mix
+real and emulated GCS usage in the same Tecken instance, so ensure that all
+the URL settings (``DJANGO_SYMBOL_URLS``, ``DJANGO_UPLOAD_DEFAULT_URL``,
+``DJANGO_UPLOAD_TRY_SYMBOLS_URL``, and ``DJANGO_UPLOAD_URL_EXCEPTIONS``)
+are consistantly set to real or emulated GCS URLs.
+
+The public host at https://storage.gcs-emulator.127.0.0.1.nip.io:4443 uses a
+self-signed certificate.  You will need to add an SSL exception when downloading
+a symbol file for the first time.
+
+Files are persisted in a volume created by ``docker-compose``. To clear old
+uploads, delete the ``gcs-emulator`` container and volumes:
+
+.. code-block:: shell
+
+    $ docker-compose rm --force --stop -v gcs-emulator
```

Comment:
Is this something we should instead put in `bin/setup-services.sh` which gets run when you do `make setup`? The thinking being that there's related data in postgres and maybe redis and it should all get wiped and rebuilt at the same time.

Reply:
If ``make setup`` means "reset the world" as well as "bootstrap my environment", then yes. It doesn't make sense for a script in the web container to call docker-compose though, I think this would be at the Makefile level.

Reply:
`make setup` isn't really about bootstrapping. It's about creating a default initial state for things including data services. That's an important step in bootstrapping a local dev environment, but it's also really handy in throwing away all the state of said environment.

We can't call docker-compose from a script inside a container--that's bad. But doing it in the `Makefile` as you suggest would be ok.